src/output.h

```diff
@@ -428,6 +428,16 @@ inline bool query_yn( const char *const msg, Args &&... args )
     return query_yn( string_format( msg, std::forward<Args>( args )... ) );
 }
 
+enum class query_ynq_result {
```

Comment:
I think your idea is great. Minor nitpick: how would you feel about consolidating "cancel" and "q"? Maybe `query_ync` instead?

Reply:
Codewise it uses the `QUIT` action and the user-facing description is `Cancel` so I feel it's fine this way.

Reply:
I was trying to keep the function name consistent with the input category/action names, so I'll probably just rename "cancel" to "quit" here. Making the code consistent with the user-facing description requires much more changes and I doubt it's worth the effort though.